test/core/surface/completion_queue_test.cc

```diff
@@ -23,6 +23,7 @@
 #include <grpc/support/time.h>
 #include "src/core/lib/gpr/useful.h"
 #include "src/core/lib/gprpp/memory.h"
+#include "src/core/lib/gprpp/sync.h"
```

Comment:
Why were all these test changes needed?

Reply:
Previously the execution of callbacks used to be inline. But since the callback is now not inline, we need to wait for the callback to complete to be able to test the results. Hence, this synchronization.

Reply:
Okay I will read the test change carefully then.